util/rom-ext-manifest-generator.py

```diff
@@ -0,0 +1,123 @@
+import copy
+from pathlib import Path
+
+import hjson
+from mako.template import Template
+
+from topgen.c import MemoryRegion, Name
+
+DESC = """ROM_EXT manifest generator"""
+
+USAGE = """
+  rom-ext-manifest-generator --input-dir:
+    Directory where manifest.hjson and manifest.h.tpl reside.
+
+  rom-ext-manifest-generator --output-dir:
+    Directory where manifest.hjson and manifest.h.tpl reside.
+"""
+
+
+def add_size_in_bytes(fields):
+    """Adds `size_in_bytes` entry to every parsed ROM_EXT manifest field.
+
+    fields[N]    =>    fields[N]
+      - ...              - ...
+      - size             - size
+                         - size_in_bytes
+    """
+
+    for field in fields:
+        field['size_in_bytes'] = field['size'] // 8
+
+
+def add_offsets_in_bytes(fields):
+    """Adds `offset_in_bytes` entry to every parsed ROM_EXT manifest field.
+
+    fields[N]    =>    fields[N]
+      - ...              - ...
+      - size_in_bytes    - size_in_bytes
+                         - offset_in_bytes
+    """
+
+    accumulated_offset = 0
+
+    for field in fields:
+        field['offset'] = hex(accumulated_offset)
+        accumulated_offset += field['size_in_bytes']
+
+
+def generate_cheader(fields, input_dir, output_dir):
+    """ Generates C header file from the `template_file`.
+
+    It produces a list of tuples with a field name and the `MemoryRegion`
+    object, whic is used in the `template_file`. The resulting header file is
+    placed into `output_dir`.
+    """
+
+    manifest_cheader_template_file = input_dir / 'manifest.h.tpl'
+    manifest_cheader_output_file = output_dir / 'manifest.h'
+
+    base_name = Name.from_snake_case("ROM_EXT")
+
+    items = []
+    for field in fields:
+        if field['type'] == "field":
+            item = (field['name'],
+                    MemoryRegion(
+                        base_name + Name.from_snake_case(field['name']),
+                        field['offset'], field['size_in_bytes']))
+            items.append(item)
+
+    with manifest_cheader_template_file.open('r') as f:
+        template = Template(f.read())
+
+    header = template.render(items=items)
+
+    with manifest_cheader_output_file.open('w') as f:
+        f.write(header)
+
+    print('Template sucessfuly written to {}.'.format(
+        manifest_cheader_output_file))
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        usage=USAGE,
+        description=DESC)
+
+    parser.add_argument('--input-dir',
+                        required=True,
+                        type=Path,
+                        help='Manifest hjson and template directory.')
+
+    parser.add_argument('--output-dir',
+                        required=True,
+                        type=Path,
+                        help='Manifest header output directory.')
+
+    args = parser.parse_args()
+
+    output_dir = args.output_dir
+    input_dir = args.input_dir
```

Comment:
I'm not sure I'd bother with the local variables. I'd probably just use `args.output_dir` and `args.input_dir` for the rest of the `main` function.

Reply:
Done.